From defaa25c2b2285587b02005ea644fb656f1655f7 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Wed, 14 Nov 2018 19:57:00 +0200
Subject: [PATCH 2/6] ethernet: ravb: Alternative PHY MDIO address might be 0

And this is exactly the Backplane board's case.
So treat invalid alternative address as "-1".

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
---
 drivers/net/ethernet/renesas/ravb_main.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/renesas/ravb_main.c b/drivers/net/ethernet/renesas/ravb_main.c
index 65a7a18..ed83e94 100644
--- a/drivers/net/ethernet/renesas/ravb_main.c
+++ b/drivers/net/ethernet/renesas/ravb_main.c
@@ -1350,7 +1350,7 @@ static int ravb_phy_start(struct net_device *ndev)
 	ret = ravb_phy_init(ndev);
 	if (ret) {
 		/* try alternative mdio address */
-		if (mdp->phy_id_alt) {
+		if (mdp->phy_id_alt != -1) {
 			mdp->phy_id = mdp->phy_id_alt;
 			ret = ravb_phy_init(ndev);
 		}
@@ -2053,7 +2053,8 @@ static int sh_mdio_init(struct ravb_private *mdp,
 		mdp->mii_bus->irq[i] = PHY_POLL;
 	if (pd->phy_irq > 0) {
 		mdp->mii_bus->irq[pd->phy] = pd->phy_irq;
-		mdp->mii_bus->irq[pd->phy_alt] = pd->phy_irq;
+		if (pd->phy_alt != -1)
+			mdp->mii_bus->irq[pd->phy_alt] = pd->phy_irq;
 	}
 
 	ret = mdiobus_register(mdp->mii_bus);
@@ -2120,7 +2121,8 @@ static struct ravb_plat_data *ravb_parse_dt(struct device *dev,
 	pdata->ether_link_active_low =
 		of_property_read_bool(np, "renesas,ether-link-active-low");
 	of_property_read_u32(np, "renesas,phy", &pdata->phy);
-	of_property_read_u32(np, "renesas,phy_alt", &pdata->phy_alt);
+	if (of_property_read_u32(np, "renesas,phy_alt", &pdata->phy_alt))
+		pdata->phy_alt = -1;
 	of_property_read_u32(np, "renesas,phy_irq", &pdata->phy_irq);
 	gpio = of_get_named_gpio_flags(np, "phy-int-gpio", 0, &flags);
 	if (gpio_is_valid(gpio)) {
-- 
2.7.4

